langchain4j-elasticsearch/src/main/java/dev/langchain4j/store/embedding/elasticsearch/ElasticsearchConfigurationKnn.java

```diff
@@ -12,23 +12,41 @@
 
 /**
  * Represents an <a href="https://www.elastic.co/">Elasticsearch</a> index as an embedding store
- * using the Knn implementation.
+ * using the approximate kNN query implementation.
+ * @see <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-knn-query.html#knn-query-top-level-parameters">kNN query</a>
  */
 public class ElasticsearchConfigurationKnn extends ElasticsearchConfiguration {
     private static final Logger log = LoggerFactory.getLogger(ElasticsearchConfigurationKnn.class);
+    private final int numCandidates;
 
     public static class Builder {
+        private int numCandidates = 0;
```

Comment:
```suggestion
        private Integer numCandidates;
```
minor: what about using `Integer` instead of `int`. This way we know if user configured a value or not (`numCandidates != null`)